script.py

```diff
@@ -0,0 +1,152 @@
+    try:
+        objects = s3.list_objects_v2(Bucket=bucket, Prefix=folder)["Contents"]
+        for obj in objects:
+            key = obj["Key"]
+            if not os.path.exists(
+                (os.path.join(s3objpath, key.split("/")[0], key.split("/")[1]))
+            ):
+                os.makedirs(
+                    os.path.join(s3objpath, key.split("/")[0], key.split("/")[1])
+                )
+            with open(
+                os.path.join(
+                    s3objpath, key.split("/")[0], key.split("/")[1], key.split("/")[3]
+                ),
+                "wb",
+            ) as f:
+                s3.download_fileobj(bucket, key, f)
+            logging.info("file with key: " + key + " downloaded from s3")
+    except Exception as e:
+        logging.error(f"Backup failed: {e}")
+
+
+def mongorestore(uri, rfolder):
+    try:
+        command = ["mongorestore", "--uri", uri, rfolder]
+        subprocess.run(command, check=True)
+        logging.info("Restore completed successfully.")
+    except subprocess.CalledProcessError as e:
+        logging.error(f"Restore failed: {e}")
+
+
+def cleanup_backups(bucket):
+    s3 = boto3.client("s3")
+    four_days_ago = datetime.now() - timedelta(days=4)
+    four_days_ago = four_days_ago.date()
+    try:
+        for i in range(1, max_backups):
+            folder = s3.list_objects_v2(Bucket=bucket, Prefix=f"{four_days_ago}_{i}")
+            if "Contents" in folder:
+                boto3.resource("s3").Bucket(bucket).objects.filter(
+                    Prefix=f"{four_days_ago}_{i}"
+                ).delete()
+            else:
+                logging.info(f"No backup exist for date {four_days_ago}_{i}.")
+    except Exception as e:
+        logging.error(f"Backup cleaning failed: {e}")
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="AWS S3 & MongoDB Script")
+    parser.add_argument("--backup", action="store_true", help="Perform backup operation")
+    parser.add_argument("--restore", action="store_true", help="Perform restore operation")
+    parser.add_argument("--cleanup", action="store_true", help="Perform cleanup operation")
+
+    args = parser.parse_args()
+
+    source_uri = os.getenv("source_uri")
+    outpath = os.getenv("outpath")
+    bucket = os.getenv("bucket")
+    # folder = os.getenv("folder")
+    s3objpath = os.getenv("s3objpath")
+    destination_uri = os.getenv("destination_uri")
+    max_backups = int(os.getenv("max_backups"))
+    # rfolder = os.getenv("restorefolder")
+
+    folder=None
+    rfolder=None
+
+    if args.backup:
+        mongodump(source_uri, outpath)
+        folder = uploadtos3(outpath, bucket, max_backups)
+    if args.restore:
+        download_from_s3(bucket, folder, s3objpath)
+        rfolder = os.path.join(s3objpath, folder)
+        mongorestore(destination_uri, rfolder)
+    if args.cleanup:
+        cleanup_backups(bucket)
+    else:
+        mongodump(source_uri, outpath)
```

Comment:
when a command is started or completed, use `logging.info` to output **in the terminal** that this command has started or completed. keep the file logs as they are